### ISSUE TITLE
Sometimes, upgrading an extension makes its license unrecognizable

### ISSUE THREAD

**Issue by LucianCheve** [xwikisas/application-licensing#165]
On two different XWiki instances, after I upgraded XWiki Pro from version 13.12 to 13.12.1, the license for XWiki Pro was no longer recognized and XWiki prompted me to buy, or input a valid license. Only after I clicked on "Check For License Updates", XWiki recognized there was already a valid license for the extension.

**Comment by AndreeaChi** [comment]
I can confirm that I got the message on a 15.5.4 XWiki that I need to add a valid license/get a trial/buy a license after I upgraded XWiki Pro to 13.12.1, even if I already had an applied license before the upgrade valid until 2027. 

After the upgrade to 13.12.1, in the Licensed Extensions table next to XWiki Pro it said - No license and it got fixed with a "Check for License Update" click.

**Comment by AlexandraSiriciuc** [comment]
The issue can be reproduced also when upgraded XWiki from 16.10.2 to 16.10.3. After the upgrade, the license for XWiki Pro was no longer recognized, all licenses were marked "No license available" .  Clicking on "Check for License updates" did not work.  Only after a restart, the issue was fixed, the license was well applied to the instance

**Comment by AlexandraSiriciuc** [comment]
The problem was reproduced again, when updating XWiki PRO, the license was no longer recognized. When adding the license code, received the message "The license is not compatible or useful for your server".  After a restart, the license was active again on the wiki

**Comment by oanalavinia** [comment]
Hi! @LucianCheve and @AlexandraSiriciuc Can you confirm this issue is random and you couldn't reproduce it all the time? I cannot reproduce this issue on a local instance so far and I would some more details. 

Configurations I used for testing:
1. XWiki 16.10.3, Jetty server. XWiki Pro 13.12. XWiki Pro was upgraded to 13.12.1,13.12.2,.. until last version. No other pro apps installed
2. XWiki 15.5.4, Jetty server. XWiki Pro 13.12. XWiki Pro was upgraded to 13.12.1,13.12.2,.. until last version. The following extensions were installed from the `XWiki Pro` administration section, with version 13.12: Calendar, Collabora, Forum, Meeting, Task Manager

So I would need more details for the use cases you mentioned (at least for the most recent ones) to know more about what was the XWiki set-up and version, or other details that might be of help. Especially for the problem encountered after an XWiki upgrade (waiting for the XWiki server and XWiki Pro version you had while encountering this issue)

**Comment by LucianCheve** [comment]
Yes, I confirm this issue happens randomly (at least it appears so, as we don't have many details about it). For the many upgrades done, for many instances with XWiki Pro (or other licenses), this issue occurred only a handful of times, and unfortunately we don't have many details about it.

 Ever since I opened this ticket, a year ago, I think I only encountered it once more.  I'm not sure about the relation between:
1. Does this happen only for the extension  "XWiki Pro" ?
2. Is it related to the fact that we upgraded the instance to a newer version?
3. Why in some cases the "Check for license updates" button fixes it, and in some other cases a restart is required?
4. Is it happening only for the cloud instances? (I don't recall having this issue on-premise, but maybe we did)

Next time we encounter this, we will try to gather as many details as we can: 

- logs 
- was nexus working? 
- history about the upgrade process of the extension
- XWiki version
- extension version - before and after upgrade
- changes to the standard flavor

**Comment by AlexandraSiriciuc** [comment]
In the issue mentioned by Clément, the client updated the Task Manager Application (Pro) to v3.10.0. After the update, they encountered the message error: "There is no valid license" ; the issue was fixed after a wiki restart.
